lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -748,15 +765,24 @@ def _set_node_conf(self, nodes: List[Node]) -> None:
                 # indicates that this unit is part of the "initial cm nodes"
                 self.peers_data.put(Scope.UNIT, "bootstrap_contributor", True)
 
-        self.opensearch_config.set_node(
-            self.app.name,
-            self.model.name,
-            self.unit_name,
-            computed_roles,
-            cm_names,
-            cm_ips,
-            contribute_to_bootstrap,
-        )
+        # Update config only when we have meaningful changes.
+        current_config = self.opensearch_config.load_node()
+        if not (
+            current_config.get("cluster.name") == f"{self.app.name}-{self.model.name}"
+            and current_config.get("node.name") == self.unit_name
+            and current_config.get("node.roles") == computed_roles
```

Comment:
This should not change from the unit itself, we have the logic that orchestrates it from the leader, it may otherwise lead to a roles not respecting our specs